### PR TITLE
add some missing NotNullWhen

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs [hunk 1]
```diff
@@ -334,7 +334,7 @@ public void Add(GitRevision revision, bool insertWithMatch = false, int insertRa
                 {
                     foreach (var parentId in parents)
                     {
-                        if (_revisionGraph.TryGetNode(parentId, out RevisionGraphRevision parentRev))
+                        if (_revisionGraph.TryGetNode(parentId, out var parentRev))
```
Comment (gerhardol) [inline GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs:337]: Why?

Comment (gerhardol) [reply GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs:337]: This is being changed from var to the actual type.
I believe this was changed by analyzers with what was possible a couple of years ago, but no generic cleanup.

**Comment by RussKie** [comment]
We kind of adopted the same approach as .NET repo use - i.e., use var only
the type is obvious. Otherwise specify the type explicitly.

**Comment by SimonCropp** [comment]
@RussKie so in this case; is the type obvious?

**Review by mstv — APPROVED** [review]
modulo use of `var`

File: GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs:337]: Here's the rule I'm referring to: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md#c-coding-style point 10.

We still need to finalise https://github.com/gitextensions/gitextensions/pull/8843 (with additional tweaks, if necessary) and update [our coding style guide](https://github.com/gitextensions/gitextensions/wiki/Guidelines%3A-Coding-Style).

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**